PublishParasoftTestResults/xsl/sarif.xsl

```diff
@@ -404,12 +405,79 @@
                 </xsl:if>
             </xsl:when>
             <xsl:otherwise>
-                <xsl:value-of select="@locFile" /><xsl:text>"</xsl:text>
+                <xsl:choose>
+                    <xsl:when test="$isCPPProReport">
+                        <!-- For cppTest professional report, "prjModule" attribute is not present.
+                            Project name presents twice in the full source path and needs to be removed.-->
+                        <xsl:value-of select="$locationMap(@locFile)" /><xsl:text>"</xsl:text>
+                    </xsl:when>
+                    <xsl:when test="/ResultsSession/@toolId = 'dottest'">
+                        <!-- For DotTest report, project name prefix is missing in "resProjPath" of "Loc".
+                               As a result, the root folder will be removed from full source path. -->
+                        <xsl:value-of select="concat('/', substring-after(@locFile, '/'))"/><xsl:text>"</xsl:text>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <!-- For Jtest and cppTest standard reports, use "resProjPath" in "Loc". -->
+                        <xsl:call-template name="processLocation">
+                            <xsl:with-param name="locNode" select="$locNode"/>
```

Comment:
Pass in "resProjPath" attribute instead of the "locNode" since we only need "resProjPath" inside the template.